server/src/completion.ts

```diff
@@ -45,12 +45,11 @@ function getPrefix(text: string, position: number): CompletionPrefix {
     let l = position - 1; // left
     for (; l >= 0 && canShift(text[l]); --l);
     let lCh = "";
-    if (l >= 0 && (text[l] == "$" || text[l] == "@" || text[l] == "%")) {
+    if (l >= 0 && (text[l] == "$" || text[l] == "@" || text[l] == "%"))
         lCh = text[l];
-    } else {
+    else
         ++l;
-    }
-    const symbol = text.substring(l, position);
+    const symbol = lCh + text.substring(l, position);
```

Comment:
When `text[l]` is a sigil, `text.substring(l, position);` will already extract the sigil. This will add it twice.
When `text[l]` is some other character, the `++l` moves the left cursor forward again.

This used to grab `text[left-1]` instead, and then would need to prepend this symbol.